### ISSUE TITLE
Cache is not invalidated when package.json changes

### ISSUE THREAD

**Issue by marechal-p** [eslint/eslint#12904]
**Tell us about your environment**

* **ESLint Version:** 6.8.0
* **Node Version:** 10.15.3
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

`@typescript-eslint/parser`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "sourceType": "module",
    "ecmaVersion": 6,
    "ecmaFeatures": {
      "jsx": true
    },
    tsconfigRootDir: __dirname,
    project: 'compile.tsconfig.json'
  },
  "plugins": [
    "@typescript-eslint",
    "@typescript-eslint/tslint",
    "import",
    "no-null"
  ],
  "env": {
    "browser": true,
    "mocha": true,
    "node": true
  },
  "ignorePatterns": [
    "node_modules",
    "*.d.ts"
  ]
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Tried to lint TS files using the cache. The problematic rule is `import/no-extraneous-dependencies`. It warns us when an implicit dependency is imported. But once I update the package.json file, the cache is not invalidated, causing the rule to warn again.

```ts
import * as meh from 'fs-extra';
```

```bash
eslint --cache source.ts
```

**What did you expect to happen?**

Modifying dependencies in the `package.json` file should invalidate eslint's cache. That or any modification to the `package.json` file.

**What actually happened? Please include the actual, raw output from ESLint.**

Cache is being used again despite update of the `package.json` file.

**Are you willing to submit a pull request to fix this bug?**

No.

**Comment by marechal-p** [comment]
Please tell me if this is a bug in the `eslint-import-plugin`, I assumed plugins had no control in what invalidates the cache.

**Comment by kaicataldo** [comment]
Dupe of https://github.com/eslint/eslint/issues/12828. We’d be happy to consider an RFC for how this could be achieved!
